src/Microsoft.AspNetCore.Mvc.ViewFeatures/Internal/ViewDataDictionaryFactory.cs

```diff
@@ -0,0 +1,16 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using Microsoft.AspNetCore.Mvc.ModelBinding;
+
+namespace Microsoft.AspNetCore.Mvc.ViewFeatures.Internal
+{
+    public class ViewDataDictionaryFactory
+    {
+        public static ViewDataDictionary CreateViewDataDictionary(IModelMetadataProvider metadataProvider)
```

Comment:
> The goal here was to not change anything that wasn't required.

Agree. In test classes that used the internal ctor they can just do `new ViewDataDictionary(metadataProvider, new ModelStateDictionary())` now instead of `ViewDataDictionaryFactory.CreateViewDataDictionary(metadataProvider)`. Does the same thing as the internal ctor did but doesn't require a factory class in the src assemblies to enable tests.